generate/generate.go

```diff
@@ -18,22 +18,31 @@ const disclaimer = "// Code generated by github.com/GannettDigital/jstransform;
 const msgpSuffix = "_msgp"
 const msgpMode = gen.Encode | gen.Decode | gen.Marshal | gen.Unmarshal | gen.Size | gen.Test
 
-// BuildArgs contains information used to build the structs for a JSONschema
-// SchemaPath is the path tot he jsonSchema file to use generate the Go struct representations
-// OutputDir is the destination for the generated files
-// GenerateAvro is a flag that defines if Avro serializing code should be built.
+// BuildArgs contains information used to build the structs for a JSONschema.
+//  SchemaPath is the path tot he jsonSchema file to use generate the Go struct representations
+//
+//  OutputDir is the destination for the generated files
+//
+//  NoNestedStructs will create structs that have no unnamed nested structs in them but rather defined types for each
+//  nested struct
+//
+//  GenerateAvro is a flag that defines if Avro serializing code should be built.
 //  The Avro generated code will only use a single field in the case where a field name is defined in a oneOf and
 //  elsewhere in the JSON schema. When converting the most specific version of such a field will be used. In general
 //  conflicting names like this should be avoided in the JSON schema.
-// GenerateMessagePack is a flag that defines if message pack serializing code should be built.
-// StructNameMap allows specifying the type name of the struct for each JSON file.
-// FieldNameMap is used to provide alternate names for fields in the resulting structs.
-//   The property names in the JSON tags for these structs remains the same as supplied.
-//   This can be used to accommodate names that are valid JSON but not valid Go identifiers
+//
+//  GenerateMessagePack is a flag that defines if message pack serializing code should be built.
+//
+//  StructNameMap allows specifying the type name of the struct for each JSON file.
+//
+//  FieldNameMap is used to provide alternate names for fields in the resulting structs.
+//  The property names in the JSON tags for these structs remains the same as supplied.
+//  This can be used to accommodate names that are valid JSON but not valid Go identifiers
 type BuildArgs struct {
 	SchemaPath             string
 	OutputDir              string
 	DescriptionAsStructTag bool
+	NoNestedStructs        bool
```

Comment:
Nit: Perhaps we could rename this to `NestedStructs` or `DisableNestedStructs` depending on what you want the default to be. `NoNestedStructs` feels a little clunky to me